### PR TITLE
fix prints and batchnorm (wrong source branch)

### PR DESCRIPTION
**Pull request by eyaler**
changed prints to functions where missing to work with python 3 and changed batchnorm function to work with tf 0.12 (see https://github.com/yenchenlin/pix2pix-tensorflow/issues/6)

### CODE REVIEW

File: model.py [hunk 1]
```diff
@@ -214,7 +222,14 @@ def discriminator(self, image, y=None, reuse=False):
 
     def generator(self, image, y=None):
         s = self.output_size
+        sb = self.output_size
+        if self.keep_aspect:
+            sb = int(sb*float(image.shape[0])/image.shape[1])
+
         s2, s4, s8, s16, s32, s64, s128 = int(s/2), int(s/4), int(s/8), int(s/16), int(s/32), int(s/64), int(s/128)
+        s2b, s4b, s8b, s16b, s32b, s64b, s128b = int(sb/2), int(sb/4), int(sb/8), int(sb/16), int(sb/32), int(sb/64), int(sb/128)
```
Comment (yenchenlin) [inline model.py:230]: Why should we use `s2b` instead of `s2`?

File: model.py [hunk 2]
```diff
@@ -214,7 +222,14 @@ def discriminator(self, image, y=None, reuse=False):
 
     def generator(self, image, y=None):
         s = self.output_size
+        sb = self.output_size
+        if self.keep_aspect:
+            sb = int(sb*float(image.shape[0])/image.shape[1])
+
         s2, s4, s8, s16, s32, s64, s128 = int(s/2), int(s/4), int(s/8), int(s/16), int(s/32), int(s/64), int(s/128)
```
Comment (yenchenlin) [inline model.py:229]: This line and the following line now have the same usage

File: main.py [hunk 3]
```diff
@@ -45,9 +47,7 @@ def main(_):
         os.makedirs(args.test_dir)
 
     with tf.Session() as sess:
-        model = pix2pix(sess, image_size=args.fine_size, batch_size=args.batch_size,
-                        output_size=args.fine_size, dataset_name=args.dataset_name,
-                        checkpoint_dir=args.checkpoint_dir, sample_dir=args.sample_dir)
+        model = pix2pix(sess, dataset_name=args.dataset_name, epochs=args.epocs, batch_size=args.batch_size, train_size=args.train_size, load_size=args.load_size, fine_size=args.fine_size, checkpoint_dir=args.checkpoint_dir, sample_dir=args.sample_dir, flips=args.flips, rotations=args.rotations, keep_aspect=args.keep_aspect)
```
Comment (yenchenlin) [inline main.py:50]: Make it follow PEP8 is better for viewing code

File: model.py [hunk 4]
```diff
@@ -10,11 +10,16 @@
 from utils import *
 
 class pix2pix(object):
-    def __init__(self, sess, image_size=256,
+    def __init__(self, sess, dataset_name='facades', epochs=args.epocs, batch_size=args.batch_size, train_size=args.train_size, load_size=args.load_size, fine_size=args.fine_size, checkpoint_dir=args.checkpoint_dir, sample_dir=args.sample_dir, flips=args.flips, rotations=args.rotations, keep_aspect=args.keep_aspect)
```
Comment (yenchenlin) [inline model.py:13]: PEP8

File: model.py [hunk 5]
```diff
@@ -151,13 +159,13 @@ def train(self, args):
...
         for epoch in xrange(args.epoch):
-            data = glob('./datasets/{}/train/*.jpg'.format(self.dataset_name))
+            data = glob('./datasets/{}/train/*.jpg'.format(self.dataset_name))+glob('./datasets/{}/train/*.png'.format(self.dataset_name))
             #np.random.shuffle(data)
             batch_idxs = min(len(data), args.train_size) // self.batch_size
 
             for idx in xrange(0, batch_idxs):
                 batch_files = data[idx*self.batch_size:(idx+1)*self.batch_size]
-                batch = [load_data(batch_file) for batch_file in batch_files]
+                batch = [load_data(batch_file, load_size=self.load_size, fine_size=self.image_size, aspect=self.keep_aspect, flip=self.flips, rot=self.rotations) for batch_file in batch_files]
```
Comment (yenchenlin) [inline model.py:168]: PEP8

File: model.py [hunk 6]
```diff
@@ -384,15 +403,15 @@ def test(self, args):
         """Test pix2pix"""
         tf.initialize_all_variables().run()
 
-        sample_files = glob('./datasets/{}/val/*.jpg'.format(self.dataset_name))
+        sample_files = glob('./datasets/{}/val/*.jpg'.format(self.dataset_name))+glob('./datasets/{}/val/*.png'.format(self.dataset_name))
```
Comment (yenchenlin) [inline model.py:406]: I think this line is rather confusing, I should make it clear that this project only support `.jpg` files.

File: model.py [hunk 7]
```diff
@@ -384,15 +403,15 @@ def test(self, args):
...
 
         # sort testing input
-        n = [int(i) for i in map(lambda x: x.split('/')[-1].split('.jpg')[0], sample_files)]
+        n = [int(i) for i in map(lambda x: x.split('/')[-1].replace('.png','jpg').split('.jpg')[0], sample_files)]
         sample_files = [x for (y, x) in sorted(zip(n, sample_files))]
 
         # load testing input
         print("Loading testing images ...")
-        sample = [load_data(sample_file, is_test=True) for sample_file in sample_files]
+        sample = [load_data(sample_file, load_size=self.load_size, fine_size=self.image_size, aspect=self.keep_aspect, flip=self.flips, rot=self.rotations, is_test=True) for sample_file in sample_files]
```
Comment (yenchenlin) [inline model.py:414]: PEP8

File: model.py [hunk 8]
```diff
@@ -384,15 +403,15 @@ def test(self, args):
         """Test pix2pix"""
         tf.initialize_all_variables().run()
 
-        sample_files = glob('./datasets/{}/val/*.jpg'.format(self.dataset_name))
+        sample_files = glob('./datasets/{}/val/*.jpg'.format(self.dataset_name))+glob('./datasets/{}/val/*.png'.format(self.dataset_name))
 
         # sort testing input
-        n = [int(i) for i in map(lambda x: x.split('/')[-1].split('.jpg')[0], sample_files)]
+        n = [int(i) for i in map(lambda x: x.split('/')[-1].replace('.png','jpg').split('.jpg')[0], sample_files)]
```
Comment (yenchenlin) [inline model.py:409]: Same here, only support `.jpg` seems cleaner.

File: utils.py [hunk 9]
```diff
@@ -17,15 +17,15 @@
 # -----------------------------
 # new added functions for pix2pix
 
-def load_data(image_path, flip=True, is_test=False):
+def load_data(image_path, load_size=286, fine_size=256, aspect=False, flip=True, rot=False, is_test=False):
     img_A, img_B = load_image(image_path)
-    img_A, img_B = preprocess_A_and_B(img_A, img_B, flip=flip, is_test=is_test)
+    img_A, img_B = preprocess_A_and_B(img_A, img_B, load_size=load_size, fine_size=fine_size, aspect=aspect, flip=flip, rot=rot, is_test=is_test)
```
Comment (yenchenlin) [inline utils.py:22]: PEP8

**Review by yenchenlin — COMMENTED** [review]
Hello @eyaler  , thanks a lot for your PR!
Please ping back after you see my review.

**Comment by eyaler** [comment]
@yenchenlin: so sorry, the last 3 commits are work in progress and not related to this pull request. Only the orignal two commits are relevant. I am not sure why they were added here. let me investigate and try to fix this.
